tests/cpp_unit_tests/test_y_bus.cpp

```diff
@@ -272,6 +280,176 @@ TEST_CASE("Test fill-in y bus") {
+                                                              14.0 + 22.0i,  26.0 + 424.0i};
+    topo.branch_bus_idx = {
+        {1, 0}, // branch 0 from node 1 to 0
+        {1, 2}, // branch 1 from node 1 to 2
+        {2, 3}, // branch 2 from node 2 to 3
+        {3, 2}, // branch 3 from node 3 to 2
+        {0, 1}, // branch 4 from node 0 to 1
+        {2, -1} // branch 5 from node 2 to "not connected"
+    };
+
+    MathModelParam<true> param_sym_update;
+    param_sym_update.branch_param = {                        //   ff,    ft,   tf,   tt
+                                     {1.0i, 0.0, 0.0, 0.0},  // {1,  0}
+                                     {0.0, 1.0, 0.0, 0.0},   // {1,  2}
+                                     {0.0, 0.0, 0.0, 2.0i},  // {2,  3}
+                                     {0.0, 0.0, 2.0, 0.0},   // {3,  2}
+                                     {0.0, 0.0, 0.0, 0.0},   // {0,  1}
+                                     {1.0i, 0.0, 0.0, 0.0}}; // {2, -1}
+    param_sym_update.shunt_param = {1.0i, 0.0i};
+
+    const ComplexTensorVector<true> admittance_sym_2 = {
+        // 17.0 + 104.0i,                                                                        [v]
+        17.0 + 105.0i, // 0, 0 -> += {1, 0}tt + {0, 1}ff + shunt(0) = 0.0 + 0.0 + 1.0i
+        // 18.0 + 3.0i,                                                                          [v]
+        18.0 + 3.0i, // 0, 1 -> += {0, 1}ft + {1, 0}tf = 0.0 + 0.0
+        // 19.0 + 2.0i,                                                                          [v]
+        19.0 + 2.0i, // 1, 0 -> += {0, 1}tf + {1, 0}ft = 0.0 + 0.0
+        // 25.0 + 1.0i,                                                                          [v]
+        25.0 + 2.0i, // 1, 1 -> += {0, 1}tt + {1, 0}ff + {1,2}ff = 0.0 + 1.0i + 0.0
+        // 6.0,                                                                                  [v]
+        7.0, // 1, 2 -> += {1,2}ft = 1.0
+        // 7.0,                                                                                  [v]
+        7.0, // 2, 1 -> += {1,2}tf = 0.0
+        // 24.0 + 1009.0i,                                                                       [v]
+        24.0 + 1010.0i, // 2, 2 -> += {1,2}tt + {2,3}ff + {3, 2}tt + {2,-1}ff = 0.0 + 0.0 + 0.0 + 1.0i
+        // 15.0 + 10.0i,                                                                         [v]
+        17.0 + 10.0i, // 2, 3 -> += {2,3}ft + {3,2}tf = 0.0 + 2.0
+        // 14.0 + 11.0i,                                                                         [v]
+        14.0 + 11.0i, // 3, 2 -> += {2,3}tf + {3,2}ft = 0.0 + 0.0
+        // 13.0 + 212.0i                                                                         [v]
+        13.0 + 214.0i // 3, 3 -> += {2,3}tt + {3,2}ff + shunt(1) = 2.0i + 0.0 + 0.0
+    };
+
+    auto VERIFY_ADMITTANCE = [](ComplexTensorVector<true> const& admittance,
```

Comment:
we reserve all-caps for C-style macros (defined using `#define`). In general, macros should only be used if there's no other choice, because they can be very dangerous. Please change this to lower case:

```suggestion
    auto verify_admittance = [](ComplexTensorVector<true> const& admittance,
```

Reply:
I did see this comment coming 😉